learners/setup.md

```diff
@@ -2,53 +2,138 @@
+
+## Computer
+
+Participants must work on a computer with a Mac, Linux, or Windows operating system (not a tablet, Chromebook, etc.) on which they have administrative privileges.
 
 ## Software Setup
 
-::::::::::::::::::::::::::::::::::::::: discussion
+To participate in this workshop, you will need to prepare the following (if you haven't already):
 
-### Details
+#### Install Shell and Git. 
+   
+:::tab
 
-Setup for different systems can be presented in dropdown menus via a `spoiler`
-tag. They will join to this discussion block, so you can give a general overview
-of the software used in this lesson here and fill out the individual operating
-systems (and potentially add more, e.g. online setup) in the solutions blocks.
+### Windows
+Git should be installed on your computer as part of your Bash install.
 
-:::::::::::::::::::::::::::::::::::::::::::::::::::
 
-:::::::::::::::: spoiler
+### Mac  
+Mac
 
-### Windows
+### Linux
+If Git is not already available on your machine you can install it via your distro's package manager. Debian/Ubuntu run 'sudo apt-get install git' and for Fedora run 'sudo dnf install git'.
 
-Use PuTTY
+:::
+  
+#### Setup GitHub 
 
-::::::::::::::::::::::::
+You will need an account for [GitHub](https://github.com/) to follow episodes 7, 8, 9 in this lesson.
 
-:::::::::::::::: spoiler
+1. To sign up for an account, navigate to https://github.com/ and follow the prompts.
+2. Verify your email address.
+3. Configure GitHub authentication.
 
-### MacOS
+You must authenticate before you can access certain resources on GitHub.
+Each way of accessing GitHub supports different authentication method. For example, you can authenticate with GitHub via the command line using Secure Shell Protocol (SSH) or you can authenticate with GitHub via browser using two-factor authentication (2FA). In our lesson we shall use both.
 
-Use Terminal.app
+:::tab
 
-::::::::::::::::::::::::
+### SSH   
 
+To set up SSH for your GitHub account, follow these steps:
 
-:::::::::::::::: spoiler
+1. Generate an SSH public/private keypair on your local machine if you don't already have one.
 
-### Linux
+To check your local machine for existing SSH keys, run the following command in the terminal:
+
+```bash
+    ls -al ~/.ssh
+```
+
+Then check the directory listing to see if you already have a public SSH key. By default, the filenames of supported public keys for GitHub are one of the following.
+
+```bash
+    id_rsa.pub
+    id_ecdsa.pub
+    id_ed25519.pub
+```
+
+If you receive an error that ~/.ssh does not exist, you do not have an existing SSH key pair in the default location. You can create a new SSH key pair in the next step. If you do have an existing SSH key pair, you can skip the next step.
+
+2. Generate a new SSH key.
+
+```bash
+ssh-keygen -t ed25519 -C "your_email@example.com"
+```
+When you're prompted to "Enter a file in which to save the key", you can press Enter to accept the default file location. 
```

Comment:
Should we include a note here about passphrases? We could flat-out recommend not using one unless requiring the extra security.

Otherwise I think we should at least indicate that if they decide to use a non-empty passphrase, they'll have to enter it **every time** they use the SSH key?